07-routing-gossip.md

```diff
@@ -173,16 +173,16 @@ The origin node:
   - MUST set `bitcoin_signature_1` and `bitcoin_signature_2` to valid
   signatures of the hash `h` (using `bitcoin_key_1` and `bitcoin_key_2`'s
   respective secrets).
-  - SHOULD set `len` to the minimum length required to hold the `features` bits
+  - SHOULD set `len` to the minimum length required to hold the `channelfeatures` bits
   it sets.
+  - MUST set a `channelfeatures` bit for each `channelfeature` negotiated for
+    the channel, according to [BOLT #9](09-features.md).
 
 The receiving node:
   - MUST verify the integrity AND authenticity of the message by verifying the
   signatures.
-  - if there is an unknown even bit in the `features` field:
-    - MUST NOT parse the remainder of the message.
+  - if there is an unknown even bit in the `channelfeatures` field:
```

Comment:
question about commit message: why would `key-exchange-instead-of-hash-preimage` be advertised as even? why is not okay to advertise it as optional and infer whatever the sender signals/uses?

Reply:
If you don't support the old style any more, you would advertize it as even, right?

Reply:
Will modify commit msg.  it was:

> Each feature can itself specify how it's advertized here: an
key-exchange-instead-of-hash-preimage feature would need to advertize
as even (you need to understand it to use it), for example, but a
wumbo feature would advertize as odd.

Now:

> Each feature can itself specify how it's advertized here: an
key-exchange-instead-of-hash-preimage feature which didn't support hash
preimages any more would need to advertize
as even (you need to understand it to use it), for example, but a
wumbo feature would advertize as odd.

Reply:
thanks for explaining, clear to me now :)